File: client/src/app/_interfaces/student-course-fee-model.ts
import { StudentPaymentModel } from "./student-payment-model";

export interface StudentCourseFeeModel {
  "order": number
  "id": string
  "priceTotal": number
  "pricePaid": number
  "priceDue": number
  "status": string
  "dateUpdate": Date
  "studentId": string
  "studentData": string
  "payments": StudentPaymentModel[]
}

File: server/src/Models/StudentCourseFeeModel.cs
/*
*@author Ramadan Ismael
*/

using System.ComponentModel.DataAnnotations;

namespace server.src.Models
{
    public class StudentCourseFeeModel
    {
        public long Order { get; set; }
        [Key]
        public string Id { get; set; } = string.Empty;

        public decimal PriceTotal { get; set; } = 0.0M;
        public decimal PricePaid { get; set; } = 0.0M;
        public decimal PriceDue { get; set; }
        public string Status { get; set; } = string.Empty;
        public DateTime? DateUpdate { get; set; }

        public string StudentId { get; set; } = string.Empty;
        public StudentDataModel? StudentData { get; set; }
        
        public ICollection<StudentPaymentModel>? Payments { get; set; }
    }
}